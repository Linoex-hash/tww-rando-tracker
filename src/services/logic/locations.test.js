import Locations from './locations';

const ITEM_LOCATIONS_FILE = {
  'Outset Island - Savage Labyrinth - Floor 30': {
    Need: 'Can Access Savage Labyrinth & Can Defeat Keese & Can Defeat Miniblins',
    'Original item': 'Triforce Chart 6',
    Types: 'Savage Labyrinth',
    Paths: ['Cave09/Room20.arc/Chest000']
  },
  "Dragon Roost Cavern - Bird's Nest": {
    Need: 'Can Access Dragon Roost Cavern & DRC Small Key x3',
    'Original item': 'Small Key',
    Types: 'Dungeon',
    Paths: ['M_NewD2/Room3.arc/Actor016']
  }
};

describe('Locations', () => {
  describe('initialize', () => {
    test('initializes the locations', () => {
      Locations.initialize(ITEM_LOCATIONS_FILE);

      expect(Locations.locations).toMatchSnapshot();
    });
  });

  describe('allLocations', () => {
    test('creates an array with all location names', () => {
      Locations.locations = {
        'Outset Island': {
          'Savage Labyrinth - Floor 30': {
            test: 'data'
          }
        },
        'Dragon Roost Cavern': {
          "Bird's Nest": {
            test: 'data'
          }
        }
      };

      expect(Locations.allLocations()).toMatchSnapshot();
    });
  });

  describe('_splitLocationName', () => {
    test('returns the general location and detailed location', () => {
      const input = 'Outset Island - Savage Labyrinth - Floor 30';

      const {
        generalLocation,
        detailedLocation
      } = Locations._splitLocationName(input);

      expect(generalLocation).toEqual('Outset Island');
      expect(detailedLocation).toEqual('Savage Labyrinth - Floor 30');
    });
  });
});
