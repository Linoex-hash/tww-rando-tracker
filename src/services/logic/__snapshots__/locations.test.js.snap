// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locations allLocations creates an array with all location names 1`] = `
Array [
  Object {
    "detailedLocation": "Savage Labyrinth - Floor 30",
    "generalLocation": "Outset Island",
  },
  Object {
    "detailedLocation": "Bird's Nest",
    "generalLocation": "Dragon Roost Cavern",
  },
]
`;

exports[`Locations initialize initializes the locations 1`] = `
Object {
  "Dragon Roost Cavern": Object {
    "Bird's Nest": Object {
      "need": "Can Access Dragon Roost Cavern & DRC Small Key x3",
      "originalItem": "Small Key",
      "types": "Dungeon",
    },
  },
  "Outset Island": Object {
    "Savage Labyrinth - Floor 30": Object {
      "need": "Can Access Savage Labyrinth & Can Defeat Keese & Can Defeat Miniblins",
      "originalItem": "Triforce Chart 6",
      "types": "Savage Labyrinth",
    },
  },
}
`;
